### ISSUE TITLE
Extracted table wrong in tabula-java 1.0

### ISSUE THREAD

**Issue by jazzido** [tabulapdf/tabula-java#155]
Hey @melisabok,

I was wondering if you could take a look at this. I tried processing [this PDF](http://www.mrh.gov.gh/files/awards/Publication_of_award_of_Bids_for_Transport_Sector__August_2016.pdf) —mentioned in tabulapdf/tabula#656.

Lattice mode works just fine in 0.9.2, but it's completely unusable in 1.0.0. To reproduce:

```
java -jar target/tabula-1.0.0-SNAPSHOT-jar-with-dependencies.jar -p 1 -a 56.925,24.255,549.945,786.555 -r Publication_of_award_of_Bids_for_Transport_Sector__August_2016.pdf
```

**Comment by melisabok** [comment]
Sure, I'll take a look.

**Comment by jazzido** [comment]
From the debug tool, these are the characters' bounding boxes with `0.9.2`:

![image](https://cloud.githubusercontent.com/assets/27584/24725851/44a9e59c-1a1e-11e7-9661-492dc2ed9d5a.png)

…and with `1.0.0`:

![image](https://cloud.githubusercontent.com/assets/27584/24725867/520f6414-1a1e-11e7-843f-4f8d287e55af.png)

Might be worth taking a look at the height calculation of `TextElement`s.

**Comment by melisabok** [comment]
You were right, the height for some TextElements was too high. I fixed it following the recommendation of this [issue](https://issues.apache.org/jira/browse/PDFBOX-3062).

The code that I added was in the LegacyPDFStreamEngine but for some reason I removed it, I don't remember why :(. 

All the tests are passing, I had to update a json file because of height differences. 


https://github.com/tabulapdf/tabula-java/pull/156
